### PR TITLE
Allow users to replace sections on personal schedule

### LINKED ISSUE #26: Allow Users to Replace Sections on Personal Schedule
Replace the section with the new one, and make the toast message reflect this, e.g. Section with enroll code xxxxx (M 9:00AM-9:50AM) replace with section with enroll code yyyyy (M 10:00AM-10:50AM).

### PR DESCRIPTION
**Pull request by scottfig**
Previously, if a user wanted to change a section, they would have to delete the previous section and add the new one from the main course page. This feature allows the user to replace an existing section linked to a course with a new section directly from the main course page. Closes #26.

deployment: https://courses-dev-realweston34.dokku-01.cs.ucsb.edu/

https://github.com/user-attachments/assets/77263f3e-5f86-4baa-b305-18b56c2c19d3

### CODE REVIEW

**Review by vzaveri23 — APPROVED** [review]
Looks good to me. The video was very helpful in allowing me to better understand the feature.

**Review by pconrad — APPROVED** [review]
Code LGTM, though I'll be honest---it's a little hard to follow.  I'll need to test the heck out of it to convince myself that it's correct
